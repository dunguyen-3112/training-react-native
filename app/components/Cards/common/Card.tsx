import {
  StyleSheet,
  View,
  ImageBackground,
  ImageSourcePropType,
  Image,
} from 'react-native';
import React, { memo } from 'react';
import { COLOR } from '@constants';
import { Button, Text } from '@components';
interface SlideItemProps {
  title: string;
  color: COLOR;
  image: ImageSourcePropType;
}

const Card = ({ image, color = 'green', title }: SlideItemProps) => {
  return (
    <ImageBackground source={image} style={styles.container}>
      <View style={styles.info}>
        <Text size={10} weight={'600'} color={color} transform="uppercase">
          Article
        </Text>
        <Text
          size={17}
          weight={'600'}
          color={'gray'}
          lineHeight={24}
          marginTop={4}
        >
          {title}
        </Text>
        <Button
          backgroundColor="green"
          paddingHorizontal={19}
          paddingVertical={6}
          marginTop={8}
        >
          <Text>Read now</Text>
          <Image
            source={require('@assets/icons/arrow.png')}
            style={styles.buttonIcon}
          />
        </Button>
      </View>
    </ImageBackground>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  container: {
    height: 169,
    width: 320,
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 32,
  },
  info: {
    maxWidth: '60%',
  },
  type: {
    textTransform: 'uppercase',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  buttonIcon: {
    marginLeft: 7,
  },
});
