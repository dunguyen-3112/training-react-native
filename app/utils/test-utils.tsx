import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
