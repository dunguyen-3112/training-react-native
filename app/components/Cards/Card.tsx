import React, { memo } from 'react';
import { StyleSheet, View, ImageBackground } from 'react-native';

import { Button, Text } from '../common';
import { ArrowICon, COLORS } from '@constants';
interface SlideItemProps {
  title: string;
  image: string;
  type: CARD_TYPES;
}

type CARD_TYPES = 'green' | 'secondary';

const Card = (props: SlideItemProps | null) => {
  const { image = '', type = 'green', title = '' } = props || {};

  return (
    <ImageBackground source={{ uri: image }} style={[styles.container]}>
      <View style={styles.info}>
        <Text fontSize="ms-0" fontWeight="600" color={type}>
          Article
        </Text>

        <Text fontSize="xl-7" fontWeight="600" customStyle={{ marginTop: 4 }}>
          {title}
        </Text>

        <Button
          paddingHorizontal={19}
          paddingVertical={6}
          customStyle={{ marginTop: 8 }}
          borderRadius={8}
          type={type}
        >
          <Text fontSize="ms" fontWeight="600" color="white">
            Read now
          </Text>
          <ArrowICon />
        </Button>
      </View>
    </ImageBackground>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  container: {
    height: 169,
    width: 320,
    justifyContent: 'space-between',
    padding: 32,
    borderRadius: 32,
  },
  info: {
    maxWidth: '60%',
    color: COLORS.LIGHT_BLACK,
  },
});
