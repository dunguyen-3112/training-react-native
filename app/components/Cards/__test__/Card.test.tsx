import { render, screen } from '@utils/test-utils';
import React from 'react';
import Card, { CardProps } from '../Card';

function setup(props?: CardProps) {
  const component = render(
    <Card
      image="https://firebasestorage.googleapis.com/v0/b/react-native-healthy-food-app.appspot.com/o/vegetale.png?alt=media&token=aec0b1a1-5f8a-4f4c-9d27-28032b2e2d79"
      title=" It is commonly used to test user interactions"
      {...(props?.type && { type: props.type })}
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
}

describe('Card', () => {
  it('Type is default', () => {
    setup();
  });
});
