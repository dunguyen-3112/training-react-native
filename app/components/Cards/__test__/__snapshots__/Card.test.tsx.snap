// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Type is default 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "borderRadius": 32,
        "height": 169,
        "justifyContent": "space-between",
        "padding": 32,
        "width": 320,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://firebasestorage.googleapis.com/v0/b/react-native-healthy-food-app.appspot.com/o/vegetale.png?alt=media&token=aec0b1a1-5f8a-4f4c-9d27-28032b2e2d79",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 169,
          "width": 320,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "color": "#330600",
        "maxWidth": "60%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1CC379",
            "fontSize": 10,
            "fontWeight": "600",
          },
          Object {},
        ]
      }
    >
      Article
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#5C5C5C",
            "fontSize": 17,
            "fontWeight": "600",
          },
          Object {
            "marginTop": 4,
          },
        ]
      }
    >
       It is commonly used to test user interactions
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#27AE60",
            "borderRadius": 8,
            "color": "#FFFFFF",
            "fontSize": 12,
            "fontWeight": "600",
            "marginTop": 8,
            "paddingHorizontal": 19,
            "paddingVertical": 6,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
              "fontWeight": "600",
            },
            Object {},
          ]
        }
      >
        Read now
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/arrow.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
